explainer.md

```diff
@@ -0,0 +1,273 @@
+# WebTransport Explainer
+
+Updated: April 2019
+
+## Problem and Motivation
+
+Many applications, such as games and live streaming, need a mechanism to send
+many messages as quickly as possibly, possibly out of order, and possibly
+unreliably from client to server or server to client.  The web platform is
+missing the capability to do this easily.
+
+Native applications can use raw UDP socekts, but those are not available on the
+web because they lack encryption and congestion control.
+
+Historically, web applications that needed bidirectional data stream between a
+client and a server could rely on WebSockets [RFC6455], a message-based
+protocol compatible with Web security model.  However, since the abstraction it
+provides is a single ordered stream of messages, it suffers from head-of-line
+blocking (HOLB), meaning that all messages must be sent and received in order
+even if they are independent and some of them are no longer needed.  This makes
+it a poor fit for latency sensitive applications which rely on partial
+reliability and stream independence for performance.
+
+We think there is a room for a simple, client-server, unoredered/unreliable API
+with minimal latency.  The WebTransport protocol provides this with a single
+transport object that closely matches the functionality provided by the QUIC
+protocol.
+
+## Goals
+
+- Provide a way to communicate with servers with low latency, including support
+for unreliable and unordered communication.
+
+- Provide an API that can be used for many use cases and network protocols,
+including both reliable and unreliable, ordered and unordered, client-server and
+p2p, data and media.
+
+## Non-goals
+
+This is not [UDP Socket API](https://www.w3.org/TR/raw-sockets/).  We must have
+encrypted and congestion-controlled communication.
+
+## Key use-cases
+
+- Receiving media pushed from server with minimal latency (out-of-order)
+
+- Sending game state with minimal latency to server in many small, unreliable,
+  out-of-order messages at a regular interval
+
+- Receiving messages pushed from server (such as notifications)
+
+- Requesting over HTTP and receiving media pushed out-of-order and unreliably
```

Comment:
I'm curious why you called out media specifically for the HTTP pooling use case. Is it worth generalizing this or calling out other use cases with HTTP pooling?

Reply:
It was the example Bernard thought of which seemed pretty good.  It's pretty common to request a media stream over HTTP.  Why not let it be a live stream that can be received out of order?  Having it be specific seems better than generic data, at least for an example.

But if you can think of a better example, I'd be happy to add it.

Reply:
sgtm